apiserver/facade/interface.go

```diff
@@ -58,6 +58,8 @@ type Context interface {
 	Resources() Resources
 
 	// XXX for a given model either State() or CAASState() will return nil
+	IsCAAS() bool
```

Comment:
You currently cannot - calling State when there is a CAASState (and vice versa) results in an intentional panic. I think adding the Is* functions is worth while, at least for now (the panics help).